### DISCUSSION TITLE
Using manual approval step

### DISCUSSION THREAD

**Discussion by tsposato** [community/community#23992] [New to GitHub]
<p>Hi,</p>
<p>I’m currently using a github action to build (via docker), and deploy my apps into ECS.<br>
Everything is now working fine deploying to our test environment on a push trigger.<br>
However, I would like a manual step on the same workflow which allows us to trigger a deployment to a pre-prod or prod environment for that specific job. We currently do this in Jenkins but I haven’t been able to find a way to do this in github actions.</p>
<p>I can see a lot of posts about <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#workflow_dispatch" rel="noopener nofollow ugc"> <code>workflow_dispatch</code> </a> but I’m not sure how I would configure it in my use case. From what I can tell this will only trigger a full workflow run and not an individual job/step which is what I require. Is this correct?</p>
<p>The branch I am using in question isn’t the master branch. I have created a separate prod-approval.yml workflow to the normal action.yml which fires off a push to said branch. But I don’t see the manual run workflow button in the Actions interface?</p>

**Comment by Ezard** [comment, marked as answer]
<p>GitHub has actually just added beta support for this :slight_smile:</p>
<p>See these documentation links for more info on how to set it up:</p>
<ul>
<li><a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/environments" rel="noopener nofollow ugc">https://docs.github.com/en/free-pro-team@latest/actions/reference/environments</a></li>
<li><a href="https://docs.github.com/en/free-pro-team@latest/actions/managing-workflow-runs/reviewing-deployments" rel="noopener nofollow ugc">https://docs.github.com/en/free-pro-team@latest/actions/managing-workflow-runs/reviewing-deployments</a></li>
</ul>

**Comment by tsposato** [comment]
<p>Looks promising, but unless I am reading it incorrectly it doesn’t work for private repositories? So people using Github for free have the functionality but paid users don’t? Seems like an odd strategy…</p>

**Comment by lgylym** [comment]
<p>Hey I cannot find any reference about not working with private repos. Where do you see it?</p>

**Comment by tsposato** [comment]
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/actions/reference/environments#about-environments" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/actions/reference/environments#about-environments" target="_blank" rel="noopener nofollow ugc">Environments - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/actions/reference/environments#about-environments" target="_blank" rel="noopener nofollow ugc">//docs.github.com/en/actions/reference/environments</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
